cmake/compilerFlags.cmake

```diff
@@ -136,4 +137,36 @@ if(MSVC)
     # Object Level Parallelism
     add_compile_options(/MP)
     add_definitions(-DNOMINMAX -DWIN32_LEAN_AND_MEAN)
+    
+    # https://stackoverflow.com/questions/44960715/how-to-enable-stdc17-in-vs2017-with-cmake
+    if (MSVC_VERSION GREATER_EQUAL "1700") # VS2015 and up

```

Comment:
I think it is wrong to add this block of code here. It is inappropriate to control the standard version with:

```
add_compile_options("/std:c++14")
```

In Modern CMake we have to use the `CMAKE_CXX_STANDARD` property as you did few days ago. Basically you are doing here what CMake is doing internally. Check the official documentation:

https://cmake.org/cmake/help/v3.17/prop_tgt/CXX_STANDARD.html

Reply:
I agree it's wrong.  However setting CMAKE_CXX_STANDARD to 11 doesn't do anything on Visual Studio.  https://stackoverflow.com/questions/44960715/how-to-enable-stdc17-in-vs2017-with-cmake

And another compilcation is that __cplusplus doesn't get set either.  So, that why I have:
```
add_compile_options("/Zc:__cplusplus")
```
which is documented here: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

Reply:
> However setting CMAKE_CXX_STANDARD to 11 doesn't do anything on Visual Studio.

Right, I know about it. It does not add anything because you do not need to do it. Even if you do not specify it, the MSVC compiler will support C++11. It is only for newer standards when you can specify those flags for MSVC. You can read more about it here:

https://devblogs.microsoft.com/cppblog/standards-version-switches-in-the-compiler/

Regarding the `add_compile_options("/Zc:__cplusplus")`, good to know. I did not know about it.

Reply:
You're right.  We don't need that code `add_compile_options("/std:c++14")`.  CMake is doing it.  However, it doesn't do `add_compile_options("/Zc:__cplusplus")`.  So __cplusplus was stuck in 1998, so I thought it was using the 1998 Standard compiler.  And I found somebody in StackOverflow saying that the whole thing was broken.

Right.  That's much better.  And I've update README.md to caution MSVC users not to use `-DCMAKE_CXX_FLAGS` (and `-DCMAKE_CXX_STANDARD`).  This is working fine on Visual Studio 2019.